galaxy_ng/app/api/v3/serializers/sync.py

```diff
@@ -127,6 +145,60 @@ class Meta:
             'ca_cert': {'write_only': True},
         }
 
+    def to_representation(self, instance):
+        """
+        Splits proxy_url field from DB in 3 fields for representation.
+        proxy_url, proxy_username, proxy_password (write-only)
+        """
+
+        if instance.proxy_url is not None:
```

Comment:
setting attributes on `instance` overwrites the properties on the underlying object, which causes issues for other methods.

I'd recommend this

```python
        data = super().to_representation(instance)

        if instance.proxy_url is not None:
            url, username, _ = strip_auth_from_url(instance.proxy_url)
            data['proxy_url'] = url
            data['proxy_username'] = username

        return data
```